Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -276,6 +276,22 @@ protected override void OnNewIntent(Intent intent)
 			await OnStateChanged();
 		}
 
+		bool CanGoBack()
+		{
+			var backPressed = BackPressed;
+
+			if (backPressed == null)
+				return true;
+
+			var invocationList = backPressed.GetInvocationList().ToList();
+
+			var canGoBack = !invocationList
+				.Select(@delegate => ((BackButtonPressedEventHandler) @delegate).Invoke(this, EventArgs.Empty))
+				.Any(b => b);
```

Comment:
Doesn't this invoke `OnBackButtonPressed` on EACH page? I don't think this is a good idea as we don't know what kind of custom logic people have in this handler. Naturally, people should expect it to be invoked when its page is at the top of the stack. Invoking handlers deeper down the stack is erroneous in my opinion. Maybe you should create a platform-specific property that allows you to override this logic according to your own use case.

Reply:
@adrianknight89 No event will not be invoked for each pages. Xamarin.Forms uses it only one times. Event will be invoked only for the last page in the stack how it did before.